sdk/python/feast/feature_store.py

```diff
@@ -1426,12 +1454,13 @@ def tqdm_builder(length):
                 tqdm_builder=tqdm_builder,
             )
 
-            self._registry.apply_materialization(
-                feature_view,
-                self.project,
-                start_date,
-                end_date,
-            )
+            if not isinstance(feature_view, OnDemandFeatureView):
```

Comment:
Rather than waiting until this line for ODFV based skip, `Continue` the looping before getting the provider (line1438).

kind of bringing uniqueness !